OBSDOCS-1041 (#75652)
openshift/openshift-docs: modules/cluster-logging-deploy-cli.adoc

  name: cluster-logging
  namespace: openshift-logging <1>
spec:
  targetNamespaces: [ ] <2>
----
<1> Ensure that the `namespace` field is set to `openshift-logging`.

> <2> For logging versions 5.7 and older, set `targetNamespaces` to `openshift-logging`. For logging versions 5.8 and later, the default installation mode is all namespaces on the cluster, as shown in the example, or specify a list of namespaces.

Reviewer: ```suggestion
<2> For logging versions 5.7 and earlier, set `targetNamespaces` to `openshift-logging`. For logging versions 5.8 and later, the default installation mode is all namespaces on the cluster, or specify a list of namespaces separated by a comma.
```
I dont think "as specified in the example is needed here because it is a callout. Also, I added seperated by a comma. I am not sure if that is accurate or not when you list out namespaces, but I think that is how I have seen it before. I think it would add some clarification. 

s/older/earlier based on ibm style.
https://www.ibm.com/docs/en/ibm-style?topic=word-usage#:~:text=your%20own%20installation.-,earlier,-adverb

Author: > <2> For logging versions 5.7 and earlier, set `targetNamespaces` to `openshift-logging`. For logging versions 5.8 and later, the default installation mode is all namespaces on the cluster, or specify a list of namespaces separated by a comma.

Seems ok to me.

Cc @r2d2rnd @anpingli

Author: @r2d2rnd @anpingli Could you please confirm if we can use "separated by a comma" as suggested [here](https://github.com/openshift/openshift-docs/pull/75652#discussion_r1666720762)? Thank you!

Reviewer 2: LGTM

Author: @opayne1 We have QE approval here for the update and so it seems we can use the "separated by a comma". Please help with merge of this PR.

  namespace: openshift-logging <1>
spec:
  targetNamespaces:

>   - openshift-logging <2>

Author: So, maybe the updated example and it`s callouts should look like following?

```
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: cluster-logging
  namespace: openshift-logging <1>
spec:
  targetNamespaces: [ ] <2>
```
<1> For logging versions 5.7 and older, the `targetNamespaces` must be `openshift-logging`. 
<2> For logging versions 5.8 and later, the default installation mode is in all namespaces on the cluster. You can leave `targetNamespaces` empty to make it available in all namespaces or specify a list of namespaces.

@r2d2rnd Could you let me know your thoughts? 

Cc @anpingli

>   - openshift-logging <2>

Reviewer 3: Hello @smunje1, 

The point <1> is wrong, I was trying to explain in https://github.com/openshift/openshift-docs/pull/75652/files#r1647380799 . 

Let me try to write from my side and you can redact as it's the best for being in the documentation.

```
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: cluster-logging
  namespace: openshift-logging <1>
spec:
  targetNamespaces: [ ] <2>
```

<1> It must be the openshift-logging namespace
<2> For logging versions 5.7 and older, the targetNamespaces must be openshift-logging.
For logging versions 5.8 and later, the default installation mode is in all namespaces on the cluster as the example or specify a list of namespaces.

----

> <1> For logging versions 5.7 and older, `ClusterLogForwarder` and `openshift-logging` custom resource can only be installed in the `openshift-logging` namespace. So, you must specify the `openshift-logging` namespace.

Author: Updated the PR as discussed [here](https://github.com/openshift/openshift-docs/pull/75652/files#r1661539798).

  targetNamespaces:
  - openshift-logging <1>
----

> <1> For logging versions 5.7 and older, the `openshift-logging` namespace must be specified. For logging versions 5.8 and later, the operator can be installed in the `openshift-logging` namespace, any other namespace, or in `All Namespaces`.

Reviewer 2: That is not correct. For logging versions 5.8 and later, you can deploy openshift-logging instance in any namespace.

Reviewer 2: apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: cluster-logging
  namespace: openshift-logging (1)
spec:
  targetNamespaces:
  - openshift-logging (2)
  
1  You must specify the `openshift-logging` namespace .
2  It is optional. You can leave it empty . that mean you can deploy openshift-logging crd in any namespace.

Author: Made the updates as discussed in [slack thread](https://redhat-internal.slack.com/archives/CB3HXM2QK/p1718710946706819).

> <1> You must specify the `openshift-logging` namespace for logging versions 5.7 and older. However, continuing to use this namespace prevents you from using the multi log forwarder feature. For logging 5.8 and later versions, you can use any namespace, as it enables the multi log forwarder feature.

Reviewer 3: Starting in RHOL 5.8, the operator can be installed in the `openshift-logging` namespace, in any other namespace/s not being  `openshift-logging`  that should be listed in the "targetNamespaces" or in **all** the namespaces. 

By default, in RHOL 5.8 or posterior versions, if it's installed from the OCP Console, it's installed in "all" namespaces.

Author: Revised sentence, "For logging versions 5.7 and older, the `openshift-logging` namespace must be specified. For logging versions 5.8 and later, the operator can be installed in the `openshift-logging` namespace, any other namespace, or in `All Namespaces`."
